library/objective-c/EnvoyHTTPStreamImpl.m

```diff
@@ -126,50 +137,59 @@ static void ios_on_complete(void *context) {
   ios_context *c = (ios_context *)context;
   EnvoyHTTPCallbacks *callbacks = c->callbacks;
   dispatch_async(callbacks.dispatchQueue, ^{
-    // TODO: release stream
-    if (atomic_load(c->canceled)) {
+    if (atomic_load(c->canceled) || !callbacks._onStreamClose) {
```

Comment:
should we allow the callbacks to not have this set? i.e if I were writing this in a server I would `ASSERT(callbacks._onStreamClose)` because it would be a programing error not to have a way to clean up this state.

Reply:
Yep that SGTM